core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,8 +321,11 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            if (deprecationLogger.get() == null) {
```

Comment:
This isn't thread safe, but since the first place to set this would be during startup, we don't have to worry about asynchronous actions, right?